### ISSUE TITLE
Admin member workshop history shows future events

### ISSUE THREAD

**Issue by biggianteye** [codebar/planner#1070]
## Description of the issue 📄

In the admin interface I was looking at the workshop history for a member to see their "no shows". The screenshot below shows two problems:

- The wording about their most recent workshop says
> The last workshop they attended was on 08 Oct 2019

but this is inaccurate as this is a future event (it's October 2nd today) and therefore they haven't attended it.
- The future event is flagged as a "no show" which negatively misrepresents this person's attendance history.

## Screenshots 📷

![Screenshot of member workshop history](https://user-images.githubusercontent.com/1482649/66036216-98575180-e504-11e9-8d73-20c7c1efdbfc.png)

## Steps to fix 🛠

- Ensure the "last attended" text only uses past events.
- Either a) don't show future events in the workshop list or b) show it and omit the symbol indicating whether or not they attended.

## To do 📋

* [ ] Claim this issue (comment below, or assign yourself if you are part of the codebar org)
* [ ] Fork and clone the repository
* [ ] Update the relevant files. Follow the steps to fix section in this issue.
* [ ] Commit your changes as one commit. Use the title of this issue as your commit message
* [ ] Submit a pull request
* [ ] Mention this issue in the PR description by including it's number
* [ ] Have your pull request reviewed & merged by a codebar team member

**Comment by notapatch** [comment]
Thanks for reporting the error it was really clear to understand with the screen print and I can confirm the bug. I've added some notes for myself, or the next programmer...

> Ensure the "last attended" text only uses past events.

The problems come from the [controller useing accepted_or_attended](https://github.com/codebar/planner/blob/master/app/controllers/admin/members_controller.rb#L10)

 ```
   def show
    ...
    @invitations = @member.workshop_invitations.accepted_or_attended  ... continues
    ...
    @last_attendance = @invitations.first.workshop if @invitations.any?
   end
```
accepted_or_attended returns members's workshop invitations where they are [attending or attended](https://github.com/codebar/planner/blob/master/app/models/concerns/invitation_concerns.rb#L14). The names are confusing and this has confused the programmer ( accepted  equals attending *scratches head* ).  So changing @last_attendance to just `accepted` would fix that problem.

Onto the table problem: 

>  a) don't show future events in the workshop

The easiest change would be to just use the attended scope as discussed in the previous problem. 

>  b) show it and omit the symbol indicating whether or not they attended

I would not recommend a missing symbol because no content can lead to formatting problems - instead a calendar icon next to attending. 

---

#### Side Issue
 [attending and attended are booleans](https://github.com/codebar/planner/blob/master/db/schema.rb#L522-L534) but the code also refers to nil - "nil booleans" need to be removed from the code base and they should have a default value instead - this would require migration changes and testing the data which will use maintainer's time.


[invitation_concern.rb](https://github.com/codebar/planner/blob/master/app/models/concerns/invitation_concerns.rb#L13-L15)
```ruby
    scope :accepted, -> { where(attending: true) }
    scope :accepted_or_attended, -> { where('attending=? or attended=?', true, true) }
    scope :not_accepted, -> { where('attending is NULL or attending = false') }
                                    die by fire ----^ 
```

### Reference
[Avoid the three state boolean problem
](https://thoughtbot.com/blog/avoid-the-threestate-boolean-problem
)

**Comment by despo** [comment]
Resolved by #1455
